sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/amqp/ConnectionHandler.java

```diff
@@ -170,7 +170,14 @@ public void onTransportError(Event event) {
         this.messagingFactory.onConnectionError(condition);
         Connection connection = event.getConnection();
         if (connection != null) {
-            connection.free();
+        	// If we don't unbind, it will cause a memory leak
+			if (connection.getTransport() != null) {
+			    connection.getTransport().unbind();
+			}
+			
+			if (connection != null) {
+			    connection.free();
```

Comment:
nit; Do we need to check again that the connection is not null, line#172 seems doing it already.

Reply:
@conniey probably something we need to evaluate for T2 as well:

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ConnectionHandler.java#L231-L233

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ConnectionHandler.java#L236-L248